src/librbd/internal.cc

```diff
@@ -2226,21 +2226,16 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     CephContext *cct = ictx->cct;
     ldout(cct, 20) << "flush " << ictx << dendl;
 
-    int r = ictx->state->refresh_if_required();
+    C_SaferCond cond;
+    io::AioCompletion *c = io::AioCompletion::create(&cond);
+    ictx->io_work_queue->aio_flush(c, false);
+
+    int r = cond.wait();
     if (r < 0) {
       return r;
     }
 
-    ictx->user_flushed();
-    C_SaferCond ctx;
-    {
-      RWLock::RLocker owner_locker(ictx->owner_lock);
-      ictx->flush(&ctx);
-    }
-    r = ctx.wait();
-
-    ictx->perfcounter->inc(l_librbd_flush);
-    return r;
+    return 0;
```

Comment:
@dillaman Was it your intention to make these flush changes be a part of this commit?

Reply:
Negative -- looks like I accidentally checked in my local tweaks to for fio from my benchmarking.